mars/oscar/backends/ray/context.pyx

```diff
@@ -0,0 +1,80 @@
+# Copyright 1999-2020 Alibaba Group Holding Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ...core cimport _Actor, ActorRef
+from ...context cimport BaseActorContext
+from ...errors import ActorNotExist, ActorAlreadyExist
+# TODO(fyrestone): Move lazy_import to oscar
+from ....utils import lazy_import
+
+ray = lazy_import("ray")
+
+
+cdef class _RayActorHandle:
+    cdef object actor_handle
+    cdef object uid
+    cdef object address
+
+
+cdef class RayActorContext(BaseActorContext):
+    cdef public dict actors
```

Comment:
I wonder how can this actor registry be visible to all actor processes? If not, ``has_actor`` may not work.

Reply:
From the discussion, mars holds a context in each node, the `actors` is not shared among them. Am I right?

Reply:
In my understanding, has_actor is needed to work in the scope of a node with processes hosting actors. I cannot see how Actor A created in Actor B can be seen in Actor C when A, B and C are all inside one node.

Reply:
It sounds like the named actor is the solution.